src/main/kotlin/org/jitsi/nlj/rtp/bandwidthestimation/GoogleCcEstimator.kt

```diff
@@ -0,0 +1,55 @@
+package org.jitsi.nlj.rtp.bandwidthestimation
+
+import java.time.Instant
+import org.jitsi.nlj.stats.NodeStatsBlock
+import org.jitsi.nlj.util.createChildLogger
+import org.jitsi.utils.logging.DiagnosticContext
+import org.jitsi.utils.logging2.Logger
+import org.jitsi_modified.impl.neomedia.rtp.remotebitrateestimator.RemoteBitrateEstimatorAbsSendTime
+import org.jitsi_modified.impl.neomedia.rtp.sendsidebandwidthestimation.SendSideBandwidthEstimation
+
+class GoogleCcEstimator(diagnosticContext: DiagnosticContext, parentLogger: Logger) : BandwidthEstimator {
+    override val algorithmName = "Google CC"
+
+    /* TODO: Use configuration service to set this default value. */
+    override var initBw: Float = 2_500_000.0f
+
+    override var minBw: Float = 30_000.0f
+
+    override var maxBw: Float = 20_000_000.0f
+
+    private val logger = parentLogger.createChildLogger(GoogleCcEstimator::class)
+
+    /**
+     * Implements the delay-based part of Google CC.
+     */
+    private var bitrateEstimatorAbsSendTime = RemoteBitrateEstimatorAbsSendTime(null, diagnosticContext, logger)
```

Comment:
seems like these should be `val`s?

Reply:
also, i'm assuming we'd want to be able to mock these out for tests and therefore would need a way to inject them.  or do you think we'd test this whole thing as black box?

Reply:
The ns3 code I'm imitating has setters for them -- I presume for a reason, so I didn't think too hard about it.  https://github.com/cisco/ns3-rmcat/blob/master/model/congestion-control/sender-based-controller.h

I'll need to change the setters to filter down to the component classes' setters.

Reply:
ok, in that case you can just get rid of the private